vllm/worker/vineyard_llm_cache.py

```diff
@@ -315,19 +335,22 @@ def prefetch_seq_kv_caches(
         duration = time.perf_counter() - start_time
         self.metrics.time_query.append(duration)
         self.metrics.normalized_time_query.append(duration/(len(query_tokens) + len(query_prefix)))
-        # If sampling is required, we need to leave one token unmatched
+        # shift
+        offset = context_len % self.chunk_size
+        matched -= offset
+        # If not comp skippable or sampling is required, we need to leave one token unmatched
         # to trigger the following sampling step in engine worker's workflow.
-        if seq_group_metadata is not None and seq_group_metadata.is_sampling_enabled:
+        if not is_comp_skippable or (
```

Comment:
This is the workaround you mentioned? We still can explore whether it's able to rebuild some seq group metadata later in this case, right? Or it's been fully evaluated and there's no chance to get performance gain here?

Reply:
we can explore it later, but my gut feeling is that this wouldn't gain many benefits.